### ISSUE TITLE
Key error when reading WSP files

### ISSUE THREAD

**Issue by psychicesp** [whitews/FlowKit#96]
EDIT:  formatting

**Describe the bug:**
I was testing your library and I keep getting a Key Error when I try to read in FlowJo WSP files.  It seems to be caused within wsp_utils._parse_wsp_samples.  When a sample does not have subpopulations it does not create a key for it in wsp_samples, which raises the key error when looping through wsp_groups['samples'] in wsp_utils.parse_wsp

**Code To Reproduce:**
```
import flowkit as fk
fks_fj = fk.Session(sample_folder)
fks_fj.import_flowjo_workspace(wsp_folder)
```
**Expected behavior:**
To import data from the flowjo workspace into the Session

**Desktop:**

OS : Windows 10 and MACOSX
Python version : 3.9.7
FlowKit version : 0.8.2

**Additional context:**

-------------------------------------------------
KeyError                                  Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_15264/170441143.py in <module>
----> 1 fks_fj.import_flowjo_workspace(wsp_path)

~\Anaconda3\lib\site-packages\flowkit\_models\session.py in import_flowjo_workspace(self, workspace_file_or_path, ignore_missing_files, ignore_transforms)
    201         :return: None
    202         """
--> 203         wsp_sample_groups = wsp_utils.parse_wsp(workspace_file_or_path, ignore_transforms=ignore_transforms)
    204         for group_name, sample_data in wsp_sample_groups.items():
    205             for sample, data_dict in sample_data.items():

~\Anaconda3\lib\site-packages\flowkit\_utils\wsp_utils.py in parse_wsp(workspace_file_or_path, ignore_transforms)
    486 
    487         for group_sample_id in group_dict['samples']:
--> 488             sample_dict = wsp_samples[group_sample_id]
    489             sample_name = sample_dict['sample_name']
    490 

**Comments:**

For my purposes this seems to be corrected by reordering the following code as such:

move:
```
        wsp_samples[sample_id] = {
            'sample_name': sample_name,
            'custom_gate_ids': set(),
            'custom_gates': [],
            'transforms': sample_xform_lut,
            'comp': sample_comp
        }
```

to before:
```
        if sample_root_sub_pop_el is None:
            continue

        sample_gates = _recurse_wsp_sub_populations(
            sample_root_sub_pop_el,
            None,  # starting at root, so no gate path
            gating_ns,
            data_type_ns
        )
```

I'm not very familiar with your library yet so I'm not confident this will not cause unwanted effects elsewhere, but it does pass all tests with this alteration.

**Comment by whitews** [comment]
Hi Eric,

Sorry for the delayed response. I believe this is similar to #91 which has been fixed in the `develop` branch. Would you mind pulling it down and verifying it is working for you?

Thanks,
Scott

**Comment by whitews** [comment]
Eric, can you verify this issue is fixed in the latest 0.9.0 release?

**Comment by psychicesp** [comment]
Scott,

Apologies for the delayed response.  It does look like the issue is fixed.  Thanks!
